packages/odyssey-react-mui/src/Select.tsx

```diff
@@ -0,0 +1,244 @@
+      value
+    );
+
+    const onChange = (
+      event: SelectChangeEvent<string | string[]>,
+      child: ReactNode
+    ) => {
+      const {
+        target: { value },
+      } = event;
+
+      // Set the field value, with some additional logic to handle array values
+      // for multi-selects
+      if (isMultiSelect) {
+        setSelectedValue(typeof value === "string" ? value.split(",") : value);
+      } else {
+        setSelectedValue(value);
+      }
+
+      // Trigger the onChange event, if one has been passed
+      if (onChangeProp) {
+        onChangeProp(event, child);
+      }
+    };
+
+    // Normalize the options array to accommodate the various
+    // data types that might be passed
+    const normalizedOptions = options.map((option) => {
+      if (typeof option === "object") {
+        return {
+          text: option.text,
+          value: option.value || option.text,
+          type: option.type === "heading" ? "heading" : "option",
+        };
+      }
+
+      return { text: option, value: option, type: "option" };
+    });
+
+    const renderValue = (selected: string | string[]) => {
+      // If the selected value isn't an array, then we don't need to display
+      // chips and should fall back to the default render behavior
+      if (typeof selected === "string") {
+        return undefined;
+      }
+
+      // Convert the selected options array into <Chip>s
+      const renderedChips = selected
+        .map((item: string) => {
+          const selectedOption = normalizedOptions.find(
+            (option) => option.value === item
+          );
+
+          if (!selectedOption) {
+            return null;
+          }
+
+          return <Chip key={item} label={selectedOption.text} />;
+        })
+        .filter(Boolean);
+
+      if (renderedChips.length === 0) {
+        return null;
+      }
+
+      // We need the <Box> to surround the <Chip>s for
+      // proper styling
+      return <Box>{renderedChips}</Box>;
+    };
+
+    // Convert the options into the ReactNode children
+    // that will populate the <Select>
+    const children = normalizedOptions.map((option) => {
+      // If it's a heading, render a heading
+      if (option.type === "heading") {
+        return <ListSubheader key={option.text}>{option.text}</ListSubheader>;
+      }
+
+      // If it's a native <Select>, render a standard HTML <option>
+      if (isNative) {
+        return (
+          <option key={option.value} value={option.value}>
+            {option.text}
+          </option>
+        );
+      }
+
+      // If it's a non-native <Select>, render a <MenuItem>
+      // If it's a multi-select, include a <Checkbox>
+      return (
+        <MenuItem key={option.value} value={option.value}>
+          {isMultiSelect && (
+            <Checkbox isChecked={selectedValue.includes(option.value)} />
+          )}
+          {option.text}
+        </MenuItem>
+      );
+    });
```

Comment:
I don't like passing `options` to components because it creates complications. I like how this helps us lock down how people use the component and use the correct components, but what about option groups?

How are we gonna solve someone passing in headers or wrappers in our `Select`? That's why it's beneficial to use the `children` method and just restrict the types of children based on the `native` prop.

Honestly, why don't we make 2 `Select` components? I was fine with them being the same, but if the types are completely different, they should be different components altogether.

Reply:
Confirmed with Lester, they still need `NativeSelect`. So we'll have to wrap that separately. It can be a separate story too.

If I remember correctly, we're using `<Select native>` rather than `<NativeSelect>` from MUI, but we can have the wrapped name be `NativeSelect`.

Reply:
I'm onboard with splitting this into two elements — `Select` and `NativeSelect`. I'm not sure about replacing `options` with `children`, though — I'd love to talk through this more.

For `Select`, it should only accept two types of children:
1. `MenuItem` — Can include a `Checkbox` if it's a multi-select
2. `ListSubheading` — For separating groups within the select

With the current method, we have full control over the component state (ie, we handle all the selection logic internally). If we expose `children`, do we need to give consumers control over the selection state too? That feels messier to me.

Reply:
What do you mean by selection state?

Reply:
Poor phrasing; I meant handling `selected={true}` manually. I don't think we do, since the current approach is working fine.

Reply:
Not strongly opinionated on either approach. Usually, I like to mimic what we do in HTML, but it doesn't matter much here.